apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java

```diff
@@ -25,51 +23,58 @@
 import com.google.gson.Gson;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Lazy;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
 
 import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
 
 @Service
 public class NamespaceService {
 
   private Logger logger = LoggerFactory.getLogger(NamespaceService.class);
   private static final Gson GSON = new Gson();
 
-  private final PortalConfig portalConfig;
-  private final PortalSettings portalSettings;
-  private final UserInfoHolder userInfoHolder;
-  private final AdminServiceAPI.NamespaceAPI namespaceAPI;
-  private final ItemService itemService;
-  private final ReleaseService releaseService;
-  private final AppNamespaceService appNamespaceService;
-  private final InstanceService instanceService;
-  private final NamespaceBranchService branchService;
-  private final RolePermissionService rolePermissionService;
-
-  public NamespaceService(
-      final PortalConfig portalConfig,
-      final PortalSettings portalSettings,
-      final UserInfoHolder userInfoHolder,
-      final AdminServiceAPI.NamespaceAPI namespaceAPI,
-      final ItemService itemService,
-      final ReleaseService releaseService,
-      final AppNamespaceService appNamespaceService,
-      final InstanceService instanceService,
-      final @Lazy NamespaceBranchService branchService,
-      final RolePermissionService rolePermissionService) {
-    this.portalConfig = portalConfig;
-    this.portalSettings = portalSettings;
-    this.userInfoHolder = userInfoHolder;
-    this.namespaceAPI = namespaceAPI;
-    this.itemService = itemService;
-    this.releaseService = releaseService;
-    this.appNamespaceService = appNamespaceService;
-    this.instanceService = instanceService;
-    this.branchService = branchService;
-    this.rolePermissionService = rolePermissionService;
-  }
+    private final PortalConfig portalConfig;
+    private final PortalSettings portalSettings;
+    private final UserInfoHolder userInfoHolder;
+    private final AdminServiceAPI.NamespaceAPI namespaceAPI;
+    private final ItemService itemService;
+    private final ReleaseService releaseService;
+    private final AppNamespaceService appNamespaceService;
+    private final InstanceService instanceService;
+    private final NamespaceBranchService branchService;
+    private final RolePermissionService rolePermissionService;
+    @Autowired
+    private NamespaceBoAsyncService namespaceBoAsyncService;
```

Comment:
Please inject this dependency as a constructor injection, as this is more friendly to unit test and is consistent with other dependencies.

Reply:
ok